compiler/damlc/tests/src/DamlcTest.hs

```diff
@@ -26,7 +29,104 @@ main = do
     defaultMain (tests damlc)
 
 tests :: FilePath -> TestTree
-tests damlc = testGroup "damlc test" $
+tests damlc = testGroup "damlc" $ map (\f -> f damlc)
+  [ testsForDamlcValidate
+  , testsForDamlcTest
+  ]
+
+testsForDamlcValidate :: FilePath -> TestTree
+testsForDamlcValidate damlc = testGroup "damlc validate-dar"
+  [ testCase "Non-existent file" $ do
+      (exitCode, stdout, stderr) <- readProcessWithExitCode damlc ["validate-dar", "does-not-exist.dar"] ""
+      exitCode @?= ExitFailure 1
+      assertInfixOf "does not exist" stderr
+      stdout @?= ""
+
+  , testCaseSteps "Good" $ \step -> withTempDir $ \projDir -> do
+      step "prepare"
```

Comment:
Generally, I wouldn’t put “trivial” stuff into its own step. Steps are mostly useful if you have expensive calls that can fail individually, e.g., first a build step and then a validation step.